long-ride-alerts/src/main/java/org/apache/flink/training/exercises/longrides/LongRidesExercise.java

```diff
@@ -98,17 +103,67 @@ public static void main(String[] args) throws Exception {
     @VisibleForTesting
     public static class AlertFunction extends KeyedProcessFunction<Long, TaxiRide, Long> {
 
+        private transient ValueState<Long> rideStartState;
```

Comment:
I would recommend to check original solution, it uses just one state
`ValueState<TaxiRide> rideState;`

Reply:
Right, I agree, their solution is a lot more clear.

I expanded that idea trying to make sure not to leak state.